rust/processor/src/gap_detector.rs

```diff
@@ -88,13 +87,13 @@ pub async fn create_gap_detector_status_tracker_loop(
         let result = match gap_detector_receiver.recv().await {
             Ok(result) => result,
             Err(e) => {
-                info!(
+                error!(
                     processor_name,
                     service_type = PROCESSOR_SERVICE_TYPE,
                     error = ?e,
                     "[Parser] Gap detector channel has been closed",
                 );
-                return;
+                panic!("[Parser] Gap detector channel has been closed");
```

Comment:
@rtso am I wrong in thinking this should be a panic if it ever happens?

Reply:
I think panic should be fine here. The gap detector channel will only be closed if the main thread is finished. There are other places, like in the fetcher channel, where we want to wait for the processor tasks to finish processing before ending the process.

Reply:
I undid this change- reasonable